### PR TITLE
Db connection java

### PR DESCRIPTION
**Pull request by mmbohari**
I have add DBConnectionJava - a java program that connects with db at Heroku and fetches results present in User_Info table.

I have also added DBConnectionJavaTest - a junit test case which tests connection with the db.

pom.xml have the junit and postgresql dependencies.

Awaiting feedback.

### CODE REVIEW

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	 * This method is used to return a connection object which is used to interact
+	 * with postgres db residing on Heroku
+	 * 
+	 * @return
+	 * @throws URISyntaxException
+	 * @throws SQLException
+	 */
+	private static Connection getConnection() throws URISyntaxException, SQLException {
+
+	    String username = "pifpprrquvvfkk";
```
Comment (abbasaliamir) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Externalize strings to external .properties file and load them via resource bundle, wherever feasible.
Altneratively, make these hardcoded strings as private static final variables. E.g.
private static final USERNAME = "pifpprrquvvfkk";

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	    return DriverManager.getConnection(dbUrl, username, password);
+	}
+
+	/**
+	 * Main method. 
+	 * 
+	 * @param args
+	 */
+	public static void main(String[] args) {
+
```
Comment (abbasaliamir) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Why leaving so many leading space lines?

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		
+		try {
+		//	Class.forName("org.postgresql.Driver");			// optional
```
Comment (abbasaliamir) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: If you don't need a line, just remove it. You can always track these in git history. Let the comment line be in the code ONLY if you really need it to be there.

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			Connection con = getConnection();
+
+			Statement stmt=con.createStatement();
```
Comment (abbasaliamir) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Try using prepared statement once you're initial set of testing is complete. Statements are prone to SQL injection attacks

File: service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+	static Connection connection = null;
+
+	/*
+	 * This method is used to create a connection object
+	 */
+	@BeforeClass
+	public static void testBeforeClass() {
+		// System.out.println("In testBeforeClass");
+		String username = "pifpprrquvvfkk";
```
Comment (abbasaliamir) [inline service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java]: same thing here as well, all your hardcoded values should become as final variables at class level

File: service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+			try {
+				connection.close();
+			} catch (SQLException e) {
+				System.err.println("Error encountered while closing connection object");
+				e.printStackTrace();
+			}
+		}
+	}
+
+}
```
Comment (abbasaliamir) [inline service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java]: what is this character after the last curly braces?

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java (hunk 1 above)
Comment (mmbohari) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Done.

I have used private static final as field modifiers for all the fields, namely: username, password, etc

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java (hunk 2 above)
Comment (mmbohari) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Removed leading blank spaces. I thought leaving a space after the method declaration makes the code look pleasing.

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java (hunk 3 above)
Comment (mmbohari) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Removed commented code lines

File: service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java (hunk 5 above)
Comment (mmbohari) [inline service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java]: Used private static final modifiers in the junit test cases as well

File: service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java (hunk 6 above)
Comment (mmbohari) [inline service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java]: It's a github error. There is no such special character

Comment (abbasaliamir) [inline service_exchange/src/test/java/imran_bohari/service_exchange/dbConnection/DBConnectionJavaTest.java]: ok

File: service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java (hunk 4 above)
Comment (mmbohari) [inline service_exchange/src/main/java/imran_bohari/service_exchange/dbConnection/DBConnectionJava.java]: Used prepared statement instead of statement

**Pull request merged**